src/app.rs

```diff
@@ -77,20 +77,28 @@ impl App {
     }
 }
 
-
 #[cfg(test)]
 mod tests {
     use super::*;
-    use tempfile::tempdir;
     use crate::args::AppArguments;
+    use std::fs;
+    use tempfile::tempdir;
 
-    #[test]
-    fn test_app_run_success() {
+    fn setup() -> (App, PathBuf) {
         let temp_dir = tempdir().unwrap();

```

Comment:
I believe that `tempdir()` creates a folder in your OS's temporary directory, but it's destroyed on `Drop`, in this function the variable `temp_dir` is dropped at the end of the scope, so I think the directory is destroyed and you're still using its path?

Assuming this is right, I don't know why this isn't causing any issues in the tests.

Reply:
its not causing any issues because App when is initialized, it will create the path in the folder that temp_dir returned.
Lets say that tempdir returned /tmp/some-path, then, when its dropped, 
App will check if /tmp/some-path exists, if not, it will create the base folder before proceeding.

note on:
https://github.com/CauaneAndrade/rust-skeleton-hexagonal/blob/bde47134d36f9596842e42510fbd65a02acf4681/src/app.rs#L55-L57

and on line 49:
https://github.com/CauaneAndrade/rust-skeleton-hexagonal/blob/bde47134d36f9596842e42510fbd65a02acf4681/src/file_manager.rs#L41-L52

Reply:
You're right! If using `tempdir()` just as a path generator, to be more clear, it might be better to use [NamedTempFile](https://docs.rs/tempfile/latest/tempfile/struct.NamedTempFile.html#method.new) (or the [Builder](https://docs.rs/tempfile/latest/tempfile/struct.Builder.html), if you need more flexibility).

Reply:
nice! will take a look at NamedTempFile :)